File: Data Structures & Algorithms/subsets/submission-9.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def backtrack(i, subset):
            if i == len(nums):
                res.append(subset[:])
                return
            
            backtrack(i + 1, subset)

            subset.append(nums[i])
            backtrack(i + 1, subset)
            subset.pop()
        
        backtrack(0, [])
        return res

File: Data Structures & Algorithms/daily-temperatures/submission-9.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Stack<int[]> stack = new Stack<>();

        for (int i = 0; i < temperatures.length; i++) {
            while (!stack.isEmpty() && temperatures[i] > stack.peek()[0]) {
                int[] temp = stack.pop();
                res[temp[1]] = i - temp[1];
            }

            stack.push(new int[] {temperatures[i], i});
        }

        return res;
    }
}
